dirty_cat/_fuzzy_join.py

```diff
@@ -187,89 +317,125 @@ def fuzzy_join(
+            "Required parameter missing: either parameter "
             "'on' or the pair 'left_on', 'right_on' should be specified."
         )
 
     if how == "left":
         main_table = left.reset_index(drop=True).copy()
         aux_table = right.reset_index(drop=True).copy()
-        main_col = left_col
-        aux_col = right_col
+        main_cols = left_col
+        aux_cols = right_col
     else:
         main_table = right.reset_index(drop=True).copy()
         aux_table = left.reset_index(drop=True).copy()
-        main_col = right_col
-        aux_col = left_col
-
-    # Make sure that the column types are string and categorical:
-    main_col_clean = main_table[main_col].astype(str)
-    aux_col_clean = aux_table[aux_col].astype(str)
+        main_cols = right_col
+        aux_cols = left_col
 
     # Warn if presence of missing values
-    if main_table[main_col].isna().any():
+    if main_table[main_cols].isna().any().any():
         warnings.warn(
-            "You are merging on missing values."
-            " The output correspondence will be random or missing."
-            " To avoid unexpected errors you can drop them.",
+            "You are merging on missing values. "
+            "The output correspondence will be random or missing. "
+            "To avoid unexpected errors you can drop them. ",
             UserWarning,
         )
 
-    all_cats = pd.concat([main_col_clean, aux_col_clean], axis=0).unique()
-
-    if encoder is None:
-        enc = HashingVectorizer(analyzer=analyzer, ngram_range=ngram_range)
-    else:
-        enc = encoder
-
-    enc_cv = enc.fit(all_cats)
-    main_enc = enc_cv.transform(main_col_clean)
-    aux_enc = enc_cv.transform(aux_col_clean)
+    main_num_cols = main_table[main_cols].select_dtypes(include="number").columns
+    aux_num_cols = aux_table[aux_cols].select_dtypes(include="number").columns
 
-    all_enc = vstack((main_enc, aux_enc))
+    # Check if included columns are numeric:
+    any_numeric = len(main_num_cols) != 0
+    # Check if included columns have mixed types:
+    mixed_types = len(main_cols) > len(main_num_cols)
 
-    tfidf = TfidfTransformer().fit(all_enc)
-    main_enc = tfidf.transform(main_enc)
-    aux_enc = tfidf.transform(aux_enc)
+    if len(main_cols) == 1 and len(aux_cols) == 1 and any_numeric is False:
+        main_cols = main_cols[0]
+        aux_cols = aux_cols[0]
 
-    # Find nearest neighbor using KNN :
-    neigh = NearestNeighbors(n_neighbors=1)
-
-    neigh.fit(aux_enc)
-    distance, neighbors = neigh.kneighbors(main_enc, return_distance=True)
-    idx_closest = np.ravel(neighbors)
+    if numerical_match in ["error"] and any_numeric:
+        raise ValueError(
+            "The columns you are trying to merge on are of numerical type. "
+            "Specify numerical_match as 'string' "
+            "or 'number'. "
+        )
+    elif numerical_match in ["number"] and any_numeric and not mixed_types:
```

Comment:
> In this scenario, choosing numerical_match="number" and having mixed types would raise an error, same for numerical_match="string".

I think it won't as there is the `elif numerical_match in ["number"] and any_numeric and mixed_types` condition.

I think we need to improve this design but I am still not sure of the way to go. Adding  `numerical_match="mixed"` would be problematic because the idea is to have something like `numerical_match is in ["number", "date", "geo"]`
but "mixed" would suppose that you infer the numerical type rather than explicitly giving it.

Either way, I think I will shortly after this PR start working on adding joins on dates, so we might discuss this further then.

Tell me what you think.